aleph-client/src/lib.rs

```diff
@@ -122,6 +124,27 @@ pub fn send_xt(connection: &Connection, xt: String, xt_name: &'static str, tx_st
     );
 }
 
+pub fn try_send_xt(
+    connection: &Connection,
+    xt: String,
+    xt_name: &'static str,
+    tx_status: XtStatus,
+) -> ApiResult<()> {
+    let block_hash = connection
+        .send_extrinsic(xt, tx_status)?
+        .ok_or_else(|| Error::Other(String::from("Could not get block hash").into()))?;
+    let block_number = connection
+        .get_header::<Header>(Some(block_hash))?
+        .expect("Block exists; qed")
+        .number;
```

Comment:
I think this isn't necessary. It's making one more RPC call to the node and not all clients of this API might want that. If they do though, they can do it themselves.

Reply:
maybe let's return the block hash from the function instead and then the consumer of this API will decide what to do with it

Reply:
if you send to mempool you won't get even that..

Reply:
Indeed, now I'm returning block number, but after waiting for it, so in case of mempool we do not gain anything, but in a single-call we get the number for free if needed